### DISCUSSION TITLE
Used but ignored property does not give an error

### DISCUSSION THREAD

**Discussion by robp94** [mapstruct/mapstruct#3556] [Q&A]
This mapping does not result in an error. Is there a config to get an error? Since payload is used and ignored, which seems wrong to me.

```
    @Mapping(target = "type", source = "payload")
    @BeanMapping(ignoreUnmappedSourceProperties = {"name", "payload"})
    public abstract Bar toBar(Foo foo);
```

We use this config 

```
@MapperConfig(
    componentModel = "cdi",
    unmappedTargetPolicy = ReportingPolicy.ERROR,
    unmappedSourcePolicy = ReportingPolicy.ERROR,
    disableSubMappingMethodsGeneration = true,
)
```

**Comment by filiphr** [comment]
Where would you expect the error? Currently, `BeanMapping#ignoreUnmappedSourceProperties` is used to see which source properties should not be treated as unmapped properties. 

Perhaps we could be a bit more strict and have an error on the `@BeanMapping` if the property is actually mapped
